homeassistant/components/ios/__init__.py

```diff
@@ -358,3 +379,85 @@ async def post(self, request: web.Request) -> web.Response:
             )
 
         return self.json({"status": "registered"})
+
+
+class iOSCarPlayConfigView(HomeAssistantView):
+    """A view that provides the CarPlay configuration."""
+
+    url = "/api/ios/carplay"
+    name = "api:ios:carplay"
+
+    async def get(self, request: web.Request) -> web.Response:
+        """Handle the GET request for CarPlay configuration."""
+        hass = request.app[KEY_HASS]
+        store = get_carplay_store(hass)
+        carplay_config = await store.async_get_data()
+        return self.json(carplay_config)
+
+
+class iOSCarPlayUpdateView(HomeAssistantView):
+    """A view that handles CarPlay configuration updates."""
+
+    url = "/api/ios/carplay/update"
+    name = "api:ios:carplay:update"
+
+    @require_admin
+    async def post(self, request: web.Request) -> web.Response:
+        """Handle the POST request for CarPlay configuration updates."""
+        try:
+            data = await request.json()
+        except ValueError:
+            return self.json_message("Invalid JSON", HTTPStatus.BAD_REQUEST)
+
+        hass = request.app[KEY_HASS]
+        store = get_carplay_store(hass)
+
+        # Validate the data structure
+        if not isinstance(data, dict):
+            return self.json_message("Data must be an object", HTTPStatus.BAD_REQUEST)
+
+        # Validate enabled field if present
+        if "enabled" in data and not isinstance(data["enabled"], bool):
```

Comment:
We can use the `RequestDataValidator` decorator with a voluptuous schema for validating the request data instead.

https://github.com/home-assistant/core/blob/8a10638470643e37a69a30aeeafe3d7d929ee023/homeassistant/components/http/data_validator.py#L21-L28

Example:
https://github.com/home-assistant/core/blob/8a10638470643e37a69a30aeeafe3d7d929ee023/homeassistant/components/backup/onboarding.py#L93-L109